File: Python/392_is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        left = right = 0
        while left < len(s) and right < len(t):
            if s[left] == t[right]:
                left +=1
            right +=1
        return True if left == len(s) else False

File: Java/392_is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int left = 0, right = 0;
        while(left < s.length() && right < t.length()){
            if(s.charAt(left) == t.charAt(right)){
                left++;
            }
            right++;
        }
        return left == s.length();
        
    }
}